In [1]:
import json
import pandas as pd

In [2]:
data = pd.read_json('sizes.json', lines=True)


In [3]:
data.head()

__v                                   _id  \
0    0  {'$oid': '557abe8565950003006fefe7'}   
1    0  {'$oid': '557abe8565950003006fefe5'}   
2    0  {'$oid': '557abe8565950003006fefe8'}   
3    0  {'$oid': '557abe8565950003006fefe6'}   
4    0  {'$oid': '557abe8565950003006fefe9'}   

                              created_at filter_label label  old_label  \
0  {'$date': '2015-06-12T11:12:05.248Z'}            S    26      [26\]   
1  {'$date': '2015-06-12T11:12:05.244Z'}           XS    24  [24, 24\]   
2  {'$date': '2015-06-12T11:12:05.249Z'}            S    28      [28\]   
3  {'$date': '2015-06-12T11:12:05.247Z'}            S    25       [25]   
4  {'$date': '2015-06-12T11:12:05.249Z'}            M    30      [30\]   

                              updated_at  
0  {'$date': '2015-07-07T16:29:11.868Z'}  
1  {'$date': '2015-07-07T16:27:03.743Z'}  
2  {'$date': '2015-07-07T16:36:22.139Z'}  
3  {'$date': '2015-07-07T16:29:03.370Z'}  
4  {'$date': '2015-07-07T16:44:25.382Z'}

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51210 entries, 0 to 51209
Data columns (total 7 columns):
__v             51210 non-null int64
_id             51210 non-null object
created_at      51210 non-null object
filter_label    294 non-null object
label           51210 non-null object
old_label       51210 non-null object
updated_at      51210 non-null object
dtypes: int64(1), object(6)
memory usage: 2.7+ MB


**Observations**
- dataframe has 7 columns/ 51210 lines
- filter_label column has only 294 entries, vs 51 210 lines
- dataframe entries are objects which is not the most suitable type for data manipulation

**Data frame columns - detailed view**

In [5]:
column_names=list(data)

In [6]:
column_types=[type(data._id[0]),
type(data.__v[0]),
type(data.created_at[0]),
type(data.filter_label[0]),
type(data.label[0]),
type(data.old_label[0]),
type(data.updated_at[0])]

In [7]:
for i in range (0,len(column_names)):
    print(column_names[i],column_types[i])

__v <class 'dict'>
_id <class 'numpy.int64'>
created_at <class 'dict'>
filter_label <class 'str'>
label <class 'str'>
old_label <class 'list'>
updated_at <class 'dict'>


** Step 1 - flatten the dataframe** 

1.1 Column **_if**: explore dictionary keys and values

In [8]:
col_name='_id'

In [9]:
def flatten_dic(df,col_name):
    keys=[]
    values=[]
    for index, row in data.iterrows():
        item=row[col_name]
        keys.extend(item.keys())
        values.extend(item.values())
    return keys, values

In [10]:
keys_id, values_id= flatten_dic(data, col_name)

In [11]:
def check_unique(item):
    unq_items=set(item)
    return len(unq_items)

In [12]:
print(check_unique(keys_id))

1


In [13]:
print(check_unique(values_id))

51210


**Observations**
- The keys are the same across all lines, therefore, can be removed, and only ID value can remain as item ID identifier

In [14]:
def clean_col(data,col_name):
    for index,row in data.iterrows():
        item=row[col_name]
        value=list(item.values())    
        data.loc[data.index[index], col_name]=str(value[0])
    return data
 


In [15]:
clean_col(data, '_id')

__v                       _id                             created_at  \
0        0  557abe8565950003006fefe7  {'$date': '2015-06-12T11:12:05.248Z'}   
1        0  557abe8565950003006fefe5  {'$date': '2015-06-12T11:12:05.244Z'}   
2        0  557abe8565950003006fefe8  {'$date': '2015-06-12T11:12:05.249Z'}   
3        0  557abe8565950003006fefe6  {'$date': '2015-06-12T11:12:05.247Z'}   
4        0  557abe8565950003006fefe9  {'$date': '2015-06-12T11:12:05.249Z'}   
5        0  557abe8565950003006ff021  {'$date': '2015-06-12T11:12:05.269Z'}   
6        0  557abe8565950003006ff025  {'$date': '2015-06-12T11:12:05.270Z'}   
7        0  557abe8565950003006ff033  {'$date': '2015-06-12T11:12:05.275Z'}   
8        0  557abe8565950003006feffa  {'$date': '2015-06-12T11:12:05.258Z'}   
9        0  557abe8565950003006feff7  {'$date': '2015-06-12T11:12:05.257Z'}   
10       0  557abe8565950003006ff036  {'$date': '2015-06-12T11:12:05.276Z'}   
11       0  557abe8565950003006ff000  {'$date': '2015-06-12T11:12:05.259Z'}   
12       0  557abe8565950003006ff050  {'$date': '2015-06-12T11:12:05.288Z'}   
13       0  557abe8565950003006feffd  {'$date': '2015-06-12T11:12:05.259Z'}   
14       0  557abe8565950003006ff001  {'$date': '2015-06-12T11:12:05.260Z'}   
15       0  557abe8565950003006ff065  {'$date': '2015-06-12T11:12:05.295Z'}   
16       0  557abe8565950003006ff00f  {'$date': '2015-06-12T11:12:05.265Z'}   
17       0  557abe8565950003006ff004  {'$date': '2015-06-12T11:12:05.260Z'}   
18       0  557abe8565950003006ff020  {'$date': '2015-06-12T11:12:05.268Z'}   
19       0  557abe8565950003006ff030  {'$date': '2015-06-12T11:12:05.274Z'}   
20       0  557abe8565950003006ff078  {'$date': '2015-06-12T11:12:05.305Z'}   
21       0  557abe8565950003006ff013  {'$date': '2015-06-12T11:12:05.266Z'}   
22       0  557abe8565950003006ff022  {'$date': '2015-06-12T11:12:05.269Z'}   
23       0  557abe8565950003006ff032  {'$date': '2015-06-12T11:12:05.275Z'}   
24       0  557abe8565950003006ff081  {'$date': '2015-06-12T11:12:05.307Z'}   
25       0  557abe8565950003006ff029  {'$date': '2015-06-12T11:12:05.272Z'}   
26       0  557abe8565950003006ff031  {'$date': '2015-06-12T11:12:05.274Z'}   
27       0  557abe8565950003006ff06c  {'$date': '2015-06-12T11:12:05.303Z'}   
28       0  557abe8565950003006ff084  {'$date': '2015-06-12T11:12:05.308Z'}   
29       0  557abe8565950003006ff02f  {'$date': '2015-06-12T11:12:05.273Z'}   
...    ...                       ...                                    ...   
51180    0  5b2bfa10df49450400fd53ff  {'$date': '2018-06-21T19:18:40.833Z'}   
51181    0  5b2bfa10df49450400fd5400  {'$date': '2018-06-21T19:18:40.852Z'}   
51182    0  5b2bfa10df49450400fd5401  {'$date': '2018-06-21T19:18:40.852Z'}   
51183    0  5b2bfa56df49450400fd5479  {'$date': '2018-06-21T19:19:50.169Z'}   
51184    0  5b2c02433ae91c04002e56c1  {'$date': '2018-06-21T19:53:39.990Z'}   
51185    0  5b2c639a5c69fa04005dcea7  {'$date': '2018-06-22T02:48:58.537Z'}   
51186    0  5b2c8aa0b826e504001c7fea  {'$date': '2018-06-22T05:35:28.448Z'}   
51187    0  5b2c8aa0b826e504001c7feb  {'$date': '2018-06-22T05:35:28.451Z'}   
51188    0  5b2c8ab323f1b504003f10e3  {'$date': '2018-06-22T05:35:47.692Z'}   
51189    0  5b2c8ab323f1b504003f10e4  {'$date': '2018-06-22T05:35:47.696Z'}   
51190    0  5b2c8c262d60830400bd1d4b  {'$date': '2018-06-22T05:41:58.769Z'}   
51191    0  5b2c8d1a23f1b504003f14d7  {'$date': '2018-06-22T05:46:02.206Z'}   
51192    0  5b2c8d50af13b60400841f27  {'$date': '2018-06-22T05:46:56.524Z'}   
51193    0  5b2c8d94af13b60400841f75  {'$date': '2018-06-22T05:48:04.464Z'}   
51194    0  5b2c8e0a5c69fa04005dee67  {'$date': '2018-06-22T05:50:02.280Z'}   
51195    0  5b2c8e102d60830400bd203f  {'$date': '2018-06-22T05:50:08.129Z'}   
51196    0  5b2c8e7e53099704006de2b6  {'$date': '2018-06-22T05:51:58.136Z'}   
51197    0  5b2c92e223f1b504003f1c5f  {'$date': '2018-06-22T06:10:42.341Z'}   
51198    0  5b2c92f35c69fa04005df4df  {'$date': '2018-06-

In [16]:
data.head()

__v                       _id                             created_at  \
0    0  557abe8565950003006fefe7  {'$date': '2015-06-12T11:12:05.248Z'}   
1    0  557abe8565950003006fefe5  {'$date': '2015-06-12T11:12:05.244Z'}   
2    0  557abe8565950003006fefe8  {'$date': '2015-06-12T11:12:05.249Z'}   
3    0  557abe8565950003006fefe6  {'$date': '2015-06-12T11:12:05.247Z'}   
4    0  557abe8565950003006fefe9  {'$date': '2015-06-12T11:12:05.249Z'}   

  filter_label label  old_label                             updated_at  
0            S    26      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1           XS    24  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2            S    28      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3            S    25       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4            M    30      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [17]:
col_name2='created_at'

In [18]:
keys_cr_at, values_cr_at=flatten_dic(data,col_name2)

In [19]:
print(check_unique(keys_cr_at))

1


In [20]:
print(check_unique(values_cr_at))

43535


In [21]:
### observation: there are 43 535 craeted_at values, less than lines in the DF, therefore, we can assume this column has nan values 

In [22]:
clean_col(data, col_name2)

__v                       _id                created_at filter_label  \
0        0  557abe8565950003006fefe7  2015-06-12T11:12:05.248Z            S   
1        0  557abe8565950003006fefe5  2015-06-12T11:12:05.244Z           XS   
2        0  557abe8565950003006fefe8  2015-06-12T11:12:05.249Z            S   
3        0  557abe8565950003006fefe6  2015-06-12T11:12:05.247Z            S   
4        0  557abe8565950003006fefe9  2015-06-12T11:12:05.249Z            M   
5        0  557abe8565950003006ff021  2015-06-12T11:12:05.269Z    0-1 years   
6        0  557abe8565950003006ff025  2015-06-12T11:12:05.270Z    0-1 years   
7        0  557abe8565950003006ff033  2015-06-12T11:12:05.275Z    1-2 years   
8        0  557abe8565950003006feffa  2015-06-12T11:12:05.258Z          NaN   
9        0  557abe8565950003006feff7  2015-06-12T11:12:05.257Z          NaN   
10       0  557abe8565950003006ff036  2015-06-12T11:12:05.276Z    1-2 years   
11       0  557abe8565950003006ff000  2015-06-12T11:12:05.259Z          NaN   
12       0  557abe8565950003006ff050  2015-06-12T11:12:05.288Z          NaN   
13       0  557abe8565950003006feffd  2015-06-12T11:12:05.259Z          NaN   
14       0  557abe8565950003006ff001  2015-06-12T11:12:05.260Z          NaN   
15       0  557abe8565950003006ff065  2015-06-12T11:12:05.295Z          NaN   
16       0  557abe8565950003006ff00f  2015-06-12T11:12:05.265Z          NaN   
17       0  557abe8565950003006ff004  2015-06-12T11:12:05.260Z          NaN   
18       0  557abe8565950003006ff020  2015-06-12T11:12:05.268Z    0-1 years   
19       0  557abe8565950003006ff030  2015-06-12T11:12:05.274Z    1-2 years   
20       0  557abe8565950003006ff078  2015-06-12T11:12:05.305Z    5-7 years   
21       0  557abe8565950003006ff013  2015-06-12T11:12:05.266Z          NaN   
22       0  557abe8565950003006ff022  2015-06-12T11:12:05.269Z    1-2 years   
23       0  557abe8565950003006ff032  2015-06-12T11:12:05.275Z    0-1 years   
24       0  557abe8565950003006ff081  2015-06-12T11:12:05.307Z   8-10 years   
25       0  557abe8565950003006ff029  2015-06-12T11:12:05.272Z    2-4 years   
26       0  557abe8565950003006ff031  2015-06-12T11:12:05.274Z    1-2 years   
27       0  557abe8565950003006ff06c  2015-06-12T11:12:05.303Z          NaN   
28       0  557abe8565950003006ff084  2015-06-12T11:12:05.308Z   8-10 years   
29       0  557abe8565950003006ff02f  2015-06-12T11:12:05.273Z    0-1 years   
...    ...                       ...                       ...          ...   
51180    0  5b2bfa10df49450400fd53ff  2018-06-21T19:18:40.833Z          NaN   
51181    0  5b2bfa10df49450400fd5400  2018-06-21T19:18:40.852Z          NaN   
51182    0  5b2bfa10df49450400fd5401  2018-06-21T19:18:40.852Z          NaN   
51183    0  5b2bfa56df49450400fd5479  2018-06-21T19:19:50.169Z          NaN   
51184    0  5b2c02433ae91c04002e56c1  2018-06-21T19:53:39.990Z          NaN   
51185    0  5b2c639a5c69fa04005dcea7  2018-06-22T02:48:58.537Z          NaN   
51186    0  5b2c8aa0b826e504001c7fea  2018-06-22T05:35:28.448Z          NaN   
51187    0  5b2c8aa0b826e504001c7feb  2018-06-22T05:35:28.451Z          NaN   
51188    0  5b2c8ab323f1b504003f10e3  2018-06-22T05:35:47.692Z          NaN   
51189    0  5b2c8ab323f1b504003f10e4  2018-06-22T05:35:47.696Z          NaN   
51190    0  5b2c8c262d60830400bd1d4b  2018-06-22T05:41:58.769Z          NaN   
51191    0  5b2c8d1a23f1b504003f14d7  2018-06-22T05:46:02.206Z          NaN   
51192    0  5b2c8d50af13b60400841f27  2018-06-22T05:46:56.524Z          NaN   
51193    0  5b2c8d94af13b60400841f75  2018-06-22T05:48:04.464Z          NaN   
51194    0  5b2c8e0a5c69fa04005dee67  2018-06-22T05:50:02.280Z          NaN   
51195    0  5b2c8e102d60830400bd203f  2018-06-22T05:50:08.129Z          NaN   
51196    0  5b2c8e7e53099704006de2b6  2018-06-22T05:51:58.136Z          NaN   
51197    0  5b2c92e223f1b504003f1c5f  2018-06-22T06:10:42.341Z          NaN   
51198    0  5b2c92f35c69fa04005df4df  2018-06-22T06:10:59

In [23]:
data.head()

__v                       _id                created_at filter_label label  \
0    0  557abe8565950003006fefe7  2015-06-12T11:12:05.248Z            S    26   
1    0  557abe8565950003006fefe5  2015-06-12T11:12:05.244Z           XS    24   
2    0  557abe8565950003006fefe8  2015-06-12T11:12:05.249Z            S    28   
3    0  557abe8565950003006fefe6  2015-06-12T11:12:05.247Z            S    25   
4    0  557abe8565950003006fefe9  2015-06-12T11:12:05.249Z            M    30   

   old_label                             updated_at  
0      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [24]:
data['created_at']  = pd.to_datetime(data['created_at'])

In [25]:
data.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

   old_label                             updated_at  
0      [26\]  {'$date': '2015-07-07T16:29:11.868Z'}  
1  [24, 24\]  {'$date': '2015-07-07T16:27:03.743Z'}  
2      [28\]  {'$date': '2015-07-07T16:36:22.139Z'}  
3       [25]  {'$date': '2015-07-07T16:29:03.370Z'}  
4      [30\]  {'$date': '2015-07-07T16:44:25.382Z'}

In [26]:
col_name3='updated_at'

In [27]:
keys_updt_at, values_updt_at=flatten_dic(data,col_name3)

In [28]:
print(check_unique(keys_updt_at))

1


In [29]:
print(check_unique(values_updt_at))

46594


In [30]:
### observation: there are 46 594 craeted_at values, less than lines in the DF, therefore, we can assume this column has nan values 

In [31]:
clean_col(data, col_name3)

__v                       _id              created_at filter_label  \
0        0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S   
1        0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS   
2        0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S   
3        0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S   
4        0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M   
5        0  557abe8565950003006ff021 2015-06-12 11:12:05.269    0-1 years   
6        0  557abe8565950003006ff025 2015-06-12 11:12:05.270    0-1 years   
7        0  557abe8565950003006ff033 2015-06-12 11:12:05.275    1-2 years   
8        0  557abe8565950003006feffa 2015-06-12 11:12:05.258          NaN   
9        0  557abe8565950003006feff7 2015-06-12 11:12:05.257          NaN   
10       0  557abe8565950003006ff036 2015-06-12 11:12:05.276    1-2 years   
11       0  557abe8565950003006ff000 2015-06-12 11:12:05.259          NaN   
12       0  557abe8565950003006ff050 2015-06-12 11:12:05.288          NaN   
13       0  557abe8565950003006feffd 2015-06-12 11:12:05.259          NaN   
14       0  557abe8565950003006ff001 2015-06-12 11:12:05.260          NaN   
15       0  557abe8565950003006ff065 2015-06-12 11:12:05.295          NaN   
16       0  557abe8565950003006ff00f 2015-06-12 11:12:05.265          NaN   
17       0  557abe8565950003006ff004 2015-06-12 11:12:05.260          NaN   
18       0  557abe8565950003006ff020 2015-06-12 11:12:05.268    0-1 years   
19       0  557abe8565950003006ff030 2015-06-12 11:12:05.274    1-2 years   
20       0  557abe8565950003006ff078 2015-06-12 11:12:05.305    5-7 years   
21       0  557abe8565950003006ff013 2015-06-12 11:12:05.266          NaN   
22       0  557abe8565950003006ff022 2015-06-12 11:12:05.269    1-2 years   
23       0  557abe8565950003006ff032 2015-06-12 11:12:05.275    0-1 years   
24       0  557abe8565950003006ff081 2015-06-12 11:12:05.307   8-10 years   
25       0  557abe8565950003006ff029 2015-06-12 11:12:05.272    2-4 years   
26       0  557abe8565950003006ff031 2015-06-12 11:12:05.274    1-2 years   
27       0  557abe8565950003006ff06c 2015-06-12 11:12:05.303          NaN   
28       0  557abe8565950003006ff084 2015-06-12 11:12:05.308   8-10 years   
29       0  557abe8565950003006ff02f 2015-06-12 11:12:05.273    0-1 years   
...    ...                       ...                     ...          ...   
51180    0  5b2bfa10df49450400fd53ff 2018-06-21 19:18:40.833          NaN   
51181    0  5b2bfa10df49450400fd5400 2018-06-21 19:18:40.852          NaN   
51182    0  5b2bfa10df49450400fd5401 2018-06-21 19:18:40.852          NaN   
51183    0  5b2bfa56df49450400fd5479 2018-06-21 19:19:50.169          NaN   
51184    0  5b2c02433ae91c04002e56c1 2018-06-21 19:53:39.990          NaN   
51185    0  5b2c639a5c69fa04005dcea7 2018-06-22 02:48:58.537          NaN   
51186    0  5b2c8aa0b826e504001c7fea 2018-06-22 05:35:28.448          NaN   
51187    0  5b2c8aa0b826e504001c7feb 2018-06-22 05:35:28.451          NaN   
51188    0  5b2c8ab323f1b504003f10e3 2018-06-22 05:35:47.692          NaN   
51189    0  5b2c8ab323f1b504003f10e4 2018-06-22 05:35:47.696          NaN   
51190    0  5b2c8c262d60830400bd1d4b 2018-06-22 05:41:58.769          NaN   
51191    0  5b2c8d1a23f1b504003f14d7 2018-06-22 05:46:02.206          NaN   
51192    0  5b2c8d50af13b60400841f27 2018-06-22 05:46:56.524          NaN   
51193    0  5b2c8d94af13b60400841f75 2018-06-22 05:48:04.464          NaN   
51194    0  5b2c8e0a5c69fa04005dee67 2018-06-22 05:50:02.280          NaN   
51195    0  5b2c8e102d60830400bd203f 2018-06-22 05:50:08.129          NaN   
51196    0  5b2c8e7e53099704006de2b6 2018-06-22 05:51:58.136          NaN   
51197    0  5b2c92e223f1b504003f1c5f 2018-06-22 06:10:42.341          NaN   
51198    0  5b2c92f35c69fa04005df4df 2018-06-22 06:10:59.751          NaN   
51199    0  5b2c92f8bff5e50400d5b890 2018-06-22 06:11:04.922          NaN   
512

In [32]:
data['updated_at']  = pd.to_datetime(data['updated_at'])

In [33]:
data.head()

__v                       _id              created_at filter_label label  \
0    0  557abe8565950003006fefe7 2015-06-12 11:12:05.248            S    26   
1    0  557abe8565950003006fefe5 2015-06-12 11:12:05.244           XS    24   
2    0  557abe8565950003006fefe8 2015-06-12 11:12:05.249            S    28   
3    0  557abe8565950003006fefe6 2015-06-12 11:12:05.247            S    25   
4    0  557abe8565950003006fefe9 2015-06-12 11:12:05.249            M    30   

   old_label              updated_at  
0      [26\] 2015-07-07 16:29:11.868  
1  [24, 24\] 2015-07-07 16:27:03.743  
2      [28\] 2015-07-07 16:36:22.139  
3       [25] 2015-07-07 16:29:03.370  
4      [30\] 2015-07-07 16:44:25.382

In [44]:
data[['old_label','filter_label','label']]

old_label filter_label  \
0                                                  [26\]            S   
1                                              [24, 24\]           XS   
2                                                  [28\]            S   
3                                                   [25]            S   
4                                                  [30\]            M   
5      [6 - 12m, 6-12 months, 6-12m, 6- 12 Months, 6 ...    0-1 years   
6      [3 - 6 months, 3 - 6m, 3-6 Months, 3-6m, EU 16...    0-1 years   
7      [2yrs, 2 years, 2y, Age 2, 2 years (92), 2 yea...    1-2 years   
8                                                   [54]          NaN   
9                                                   [48]          NaN   
10                                        [12-24 months]    1-2 years   
11                                                   [C]          NaN   
12                                                [sand]          NaN   
13     [Pink, fluorescent pink, pink grapefruit, Pink...          NaN   
14                                                   [D]          NaN   
15                                          [Anthracite]          NaN   
16                      [N, UK N, Size N, 53, Large N-O]          NaN   
17                                       [Full alphabet]          NaN   
18     [12-18m, 12 - 18m, 12 - 18 Months, 12-18 month...    0-1 years   
19     [18-24m, 18 - 24m, 18 - 24 Months, 18-24, 18-2...    1-2 years   
20     [6 years, Age 6, 6y, 6 years (116), 6 years (1...    5-7 years   
21                                             [R, UK R]          NaN   
22     [1 - 2yrs, 1-2 years, 1- 2yrs, 1 - 2 years, 1-...    1-2 years   
23                                      , 12 Months]    0-1 years   
24     [7 - 8 years, 7 - 8yrs, 7-8 yrs, 7-8 years, 7-...   8-10 years   
25     [2-3 years, 2 - 3yrs, 2-3y, 2 - 3 Years, Age 2...    2-4 years   
26                                     [UK 4.5 / EUR 21]    1-2 years   
27                                                [Teal]          NaN   
28           [8 years, Age 8, 8y, 8 years (128), 8years]   8-10 years   
29       [6-9 months, 6 - 9 months, 6 - 9m, 6-9 m, 6-9m]    0-1 years   
...                                                  ...          ...   
51180                                                 []          NaN   
51181                                                 []          NaN   
51182                                                 []          NaN   
51183                                                 []          NaN   
51184                                                 []          NaN   
51185                                                 []          NaN   
51186                                                 []          NaN   
51187                                                 []          NaN   
51188                                                 []          NaN   
51189                                                 []          NaN   
51190                                                 []          NaN   
51191                                                 []          NaN   
51192                                                 []          NaN   
51193                                                 []          NaN   
51194                                                 []          NaN   
51195                                                 []          NaN   
51196                                                 []          NaN   
51197                                                 []          NaN   
51198                                                 []          NaN   
51199                                                 []          NaN   
51200                                                 []          NaN   
51201                                                 []          NaN   
51202                                                 []          NaN   
51203                               

In [ ]:
#tidying up old_label
## create a sentence of words separated by a comma
## create word corpus of every appearing word

In [80]:
for index, row in data.iterrows():
    item=row['old_label']
    print(', '.join(item))
    



26\
24, 24\
28\
25
30\
6 - 12m, 6-12 months, 6-12m, 6- 12 Months, 6 - 12 Months, 6-12 moths, 74 - 80 months (74 - 80cm), UK size 2-3 (6-12 months), EU 20 - 21 (6 - 12 months), EU 18 - 19 (6 - 12 months), UK 2-3 (6-12 months), 6-12 month, UK size 2-3(6-12 months), 6-12m (-£2.00)
3 - 6 months, 3 - 6m, 3-6 Months, 3-6m, EU 16 - 17 (3 - 6 months), 3-6 m, 18 - 19 (3 - 6 months), EU 18 - 19 (3 - 6 months), UK size 1.5 (3-6 months), UK size 1.5(3-6 months)
2yrs, 2 years, 2y, Age 2, 2 years (92), 2 years (92cm)
54
48
12-24 months
C
sand
Pink, fluorescent pink, pink grapefruit, Pink Flag
D
Anthracite
N, UK N, Size N, 53, Large N-O
Full alphabet
12-18m, 12 - 18m, 12 - 18 Months, 12-18 months, UK size 4-5 (12-18 months), EU 22 - 23 (12 - 18 years), EU 20 - 21 (12 - 18 months), 12 -18 months, 12-18 m, 12-18m (+£2.00), UK 4-5 (12-18 months), UK size 4-5 (12-18months), UK size 4-5(12-18 months), 12- 18 Months, 12-18, 12-18 months (size 4-5)
18-24m, 18 - 24m, 18 - 24 Months, 18-24, 18-24m (+£2.00)

30FF
10.5, EU 45
light
Taurus
6-8
51-17
48-19
34F
scarlet
Baby Pink
49-20
34A
Srinagar
sky blue
Green Opal
Anise Etoile
UK 9.5, 9.5
Cassis
Decorative
53-18
Dark Red
Regular
36D
UK I 1/2
45-20
56 ( UK size P 1/2 )
semi circle
UK Q 1/2
50-19
UK 11, 11 (45), UK11
Jet Black
Curved
!
TWIST A TWILL PETROLIUM
Gold Link
Monochrome
55-14
Mini UK 3, 36IT, UK 3, UK 3 (EU 36), UK3, UK3 (EU36)
Chalk White
Matt Gold
Cafelatte
Dusty blue
UK E 1/2
46-26
Oat, Beige
49-21
Gold Icosahedron
Gold Tetrahedron
Keys
Peridot Opal
Manoir
58
Gold Dodecahedron
camel
Stick
Blue Opal
56-17
Aquamarine
49-23
61
58-14
Gold Octahedron
Cafe Moussins
Aubergine

Spotty
Satin Black
34 - 34, 34:34
Grey & Pink
30S
36S
34R
White & Blue
27:28
Marzipan & Candied Peel
Green & Red
32S
27:32

30R
36 X 32, 36\/32\, 36x32, W 36 / L 32, 36:32
28:28
Wax Camo
Silver Link & Silver Chain
28:32
Ylang Ylang & Patchouli
Teal & Pink
Zinc
38R
38\/32\, W 38 / L 32
Red & Green
36L
38S
34S
Neon Orange & White
Limeloaf & Ginger
34L
31:32

38L
Gre

In [86]:
for index, row in data.iterrows():
    item=row['old_label']
    print (type(item))
    break

<class 'list'>


In [71]:
from sklearn.feature_extraction.text import TfidfVectorizer

In [72]:
vectorizer = TfidfVectorizer()
#tfidf = TfidfVectorizer(tokenizer=tokenize, stop_words='english')
#docs = data['old_label']
#tfs = tfidf.fit_transform(docs)
foo=vectorizer.fit_transform(data.old_label)

AttributeError: 'list' object has no attribute 'lower'

In [73]:
import re
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords
from nltk.stem.wordnet import WordNetLemmatizer
from nltk.stem.porter import PorterStemmer

In [76]:
import nltk

nltk.download()

showing info https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/index.xml


True

In [78]:
all_tokens=[]
for 
data['tokenized_text'] = data['old_label'].apply(word_tokenize) 

TypeError: expected string or bytes-like object

In [79]:
for index,row in data.iterrows():
    item=row['old_label']
    print(item)
    break


['26\\']
